### ISSUE TITLE
[Bug]: defaultSession.webRequest.onHeadersReceived not being fired once app is packaged

### ISSUE THREAD

**Issue by willie-yao** [electron/electron#37762]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

20.0.2

### What operating system are you using?

Windows

### Operating System Version

Windows 11

### What arch are you using?

x64

### Last Known Working Electron version

20.0.2

### Expected Behavior

defaultSession.webRequest.onHeadersReceived fires both when app is packaged and in development

### Actual Behavior

defaultSession.webRequest.onHeadersReceived does not fire when the app is packaged.

Link to repo and place in code: https://github.com/willie-yao/tournament-vod-clipper/blob/main/src/main/main.ts#L406

### Testcase Gist URL
https://gist.github.com/willie-yao/40d31f2d095c521a15a89b6daacf5706

[Branch with simple test case](https://github.com/willie-yao/tournament-vod-clipper/tree/test-case)
This branch adds a `throw` statement to the `onHeadersReceived` to show if it's called when the app is packaged, as well as loading a Twitch embed immediately.

- Clone this branch
- Run npm install + npm start
- Observe an error from the throw when trying to load a Twitch embed
- Package app with `npm run package`
- When trying to load the Twitch embed, `onHeadersReceived` is never called so no throw occurs.

### Additional Information

I am writing a tool that plays an embedded Twitch video. This video works fine in development with a `localhost` server but does not work when the app is packaged. I observed that the headers were not properly removed when the app is packaged because defaultSession.webRequest.onHeadersReceived is never called.

This is similar to #23485

**Comment by Medic118** [comment]
im having the same problem. i believe it has something to do with Windows 11, i dont actually know how to code so idk if it is or how to fix it but i do know that this also isnt working for me on Windows 11.

**Comment by willie-yao** [comment]
@VerteDinde I tried reproducing this in a gist but it works in the gist. I observed that both the gist and my packaged app are using the "file://" directive. Is there something I'm missing in my code? I ensured that I'm on the same electron version.

Code in my project: https://github.com/willie-yao/tournament-vod-clipper/blob/f2825081e3e44fda6ea10516f235e137dfa83812/src/main/main.ts#L405
Gist: https://gist.github.com/willie-yao/40d31f2d095c521a15a89b6daacf5706
